api/src/main/java/org/openmrs/api/UserService.java

```diff
@@ -102,6 +102,25 @@
 	@Authorized( { PrivilegeConstants.GET_USERS })
 	public User getUserByUsername(String username) throws APIException;
 	
+	
+	/**
+	 * @param email 
+	 * @return requested user
+	 * @throws APIException
+	 */
+	@Authorized( { PrivilegeConstants.GET_USERS })
+	public User getUserByEmail(String email) throws APIException;
+	
+	/**
+	 * gets the user with the specified username or email address
+	 * 
+	 * @param email Or username to look up user with
+	 * @return requested user
+	 * @throws APIException
+	 */
+	@Authorized( { PrivilegeConstants.GET_USERS })
+	public User getUserByEmailOrUsername(String emailOrName) throws APIException;
```

Comment:
I don't think you need this since you already have `getUserByUsername` and you've added `getUserByEmail`.

Reply:
I was thinking of a single query to hit the db with an or conditioning since i found out from the validation file that email were accepted as username and instead of calling two different methods when i either an email or password is supplied for reseting password i just call this generic method to get the user then i now use the email associated with the user.

Reply:
Which validation file?

Reply:
here is the file i saw. https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/validator/UserValidator.java#L179

Reply:
Username and email are different, I don't think I understand what you mean